### PR TITLE
test: added dataplane cluster service handler unit tests

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
test: added dataplane cluster service handler unit tests

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- ~~[ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (machi1990) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:240]: NodeInfo
Total
Remaining
ResizeInfo

are being removed in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1068 we might want to remove the test case for them as they'll be removed anytime soon depending on when the linked PR is merged

Comment (pawelpaszki) [reply internal/kafka/internal/handlers/data_plane_cluster_test.go:240]: thanks for your review @machi1990. I will hold off with updating this PR after the #1068 is merged. Otherwise the tests will fail

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (ziccardi) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:79]: If expect fails, the Body is not closed. This should be moved before the Expect or `defer` should be used.

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (ziccardi) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:129]: See previous comment

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (ziccardi) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:73]: Let's add a `t.Parallel()` call here for parallel test execution

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (ziccardi) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:123]: Let's add a `t.Parallel()` call here for parallel test execution

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (ziccardi) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:185]: Let's add a `t.Parallel()` call here for parallel test execution

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (ziccardi) [inline internal/kafka/internal/handlers/data_plane_cluster_test.go:281]: Let's add a `t.Parallel()` call here for parallel test execution

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (pawelpaszki) [reply internal/kafka/internal/handlers/data_plane_cluster_test.go:79]: done

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (pawelpaszki) [reply internal/kafka/internal/handlers/data_plane_cluster_test.go:130]: done

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (pawelpaszki) [reply internal/kafka/internal/handlers/data_plane_cluster_test.go:188]: done

File: internal/kafka/internal/handlers/data_plane_cluster_test.go
Comment (pawelpaszki) [reply internal/kafka/internal/handlers/data_plane_cluster_test.go:285]: done

File: internal/kafka/internal/services/data_plane_cluster.go
Comment (machi1990) [inline internal/kafka/internal/services/data_plane_cluster.go:21]: good catch!

**Review by machi1990 — APPROVED** [review]
lgtm

I'll let @ziccardi do another round of review as well

**Pull request merged**
